### ISSUE TITLE
pythonPackages.psycopg2 build fails on Darwin

### ISSUE THREAD

**Issue by copumpkin** [NixOS/nixpkgs#11394]
```
...
clang -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -DPSYCOPG_DEFAULT_PYDATETIME=1 -DPSYCOPG_VERSION="2.5.4 (dt dec pq3 ext)" -DPSYCOPG_EXTENSIONS=1 -DPSYCOPG_NEW_BOOLEAN=1 -DHAVE_PQFREEMEM=1 -DPG_VERSION_HEX=0x090405 -DPSYCOPG_EXTENSIONS=1 -DPSYCOPG_NEW_BOOLEAN=1 -DHAVE_PQFREEMEM=1 -I/nix/store/ja6r5fd540kgz4qm7q2lrs0v379msbdx-python-2.7.10/include/python2.7 -I. -I/nix/store/wlva88arppr01gbgsc970kv5f0c22vv9-postgresql-9.4.5/include -I/nix/store/wlva88arppr01gbgsc970kv5f0c22vv9-postgresql-9.4.5/include/server -c psycopg/microprotocols_proto.c -o build/temp.macosx-10.7-x86_64-2.7/psycopg/microprotocols_proto.o
In file included from psycopg/microprotocols_proto.c:27:
In file included from ./psycopg/psycopg.h:33:
./psycopg/config.h:71:13: warning: unused function 'Dprintf' [-Wunused-function]
static void Dprintf(const char *fmt, ...) {}
            ^
1 warning generated.
clang -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -DPSYCOPG_DEFAULT_PYDATETIME=1 -DPSYCOPG_VERSION="2.5.4 (dt dec pq3 ext)" -DPSYCOPG_EXTENSIONS=1 -DPSYCOPG_NEW_BOOLEAN=1 -DHAVE_PQFREEMEM=1 -DPG_VERSION_HEX=0x090405 -DPSYCOPG_EXTENSIONS=1 -DPSYCOPG_NEW_BOOLEAN=1 -DHAVE_PQFREEMEM=1 -I/nix/store/ja6r5fd540kgz4qm7q2lrs0v379msbdx-python-2.7.10/include/python2.7 -I. -I/nix/store/wlva88arppr01gbgsc970kv5f0c22vv9-postgresql-9.4.5/include -I/nix/store/wlva88arppr01gbgsc970kv5f0c22vv9-postgresql-9.4.5/include/server -c psycopg/typecast.c -o build/temp.macosx-10.7-x86_64-2.7/psycopg/typecast.o
clang -bundle -undefined dynamic_lookup build/temp.macosx-10.7-x86_64-2.7/psycopg/psycopgmodule.o build/temp.macosx-10.7-x86_64-2.7/psycopg/green.o build/temp.macosx-10.7-x86_64-2.7/psycopg/pqpath.o build/temp.macosx-10.7-x86_64-2.7/psycopg/utils.o build/temp.macosx-10.7-x86_64-2.7/psycopg/bytes_format.o build/temp.macosx-10.7-x86_64-2.7/psycopg/connection_int.o build/temp.macosx-10.7-x86_64-2.7/psycopg/connection_type.o build/temp.macosx-10.7-x86_64-2.7/psycopg/cursor_int.o build/temp.macosx-10.7-x86_64-2.7/psycopg/cursor_type.o build/temp.macosx-10.7-x86_64-2.7/psycopg/diagnostics_type.o build/temp.macosx-10.7-x86_64-2.7/psycopg/error_type.o build/temp.macosx-10.7-x86_64-2.7/psycopg/lobject_int.o build/temp.macosx-10.7-x86_64-2.7/psycopg/lobject_type.o build/temp.macosx-10.7-x86_64-2.7/psycopg/notify_type.o build/temp.macosx-10.7-x86_64-2.7/psycopg/xid_type.o build/temp.macosx-10.7-x86_64-2.7/psycopg/adapter_asis.o build/temp.macosx-10.7-x86_64-2.7/psycopg/adapter_binary.o build/temp.macosx-10.7-x86_64-2.7/psycopg/adapter_datetime.o build/temp.macosx-10.7-x86_64-2.7/psycopg/adapter_list.o build/temp.macosx-10.7-x86_64-2.7/psycopg/adapter_pboolean.o build/temp.macosx-10.7-x86_64-2.7/psycopg/adapter_pdecimal.o build/temp.macosx-10.7-x86_64-2.7/psycopg/adapter_pint.o build/temp.macosx-10.7-x86_64-2.7/psycopg/adapter_pfloat.o build/temp.macosx-10.7-x86_64-2.7/psycopg/adapter_qstring.o build/temp.macosx-10.7-x86_64-2.7/psycopg/microprotocols.o build/temp.macosx-10.7-x86_64-2.7/psycopg/microprotocols_proto.o build/temp.macosx-10.7-x86_64-2.7/psycopg/typecast.o -L/nix/store/ja6r5fd540kgz4qm7q2lrs0v379msbdx-python-2.7.10/lib -L/nix/store/wlva88arppr01gbgsc970kv5f0c22vv9-postgresql-9.4.5/lib -lpq -lssl -lcrypto -o build/lib.macosx-10.7-x86_64-2.7/psycopg2/_psycopg.so
ld: library not found for -lssl
clang-3.6: error: linker command failed with exit code 1 (use -v to see invocation)
```

Not sure why `libssl` is failing only on darwin and don't have time to check right now, but I figured I'd flag it.

**Comment by domenkozar** [comment]
cc @datakurre @FRidh

**Comment by copumpkin** [comment]
Adding the following to the `psycopg2` expression seems to fix it on Darwin, but I'm still not sure why it works on Linux or whether it's revealing some other weirdness about our Darwin environment.

```nix
{
  # ...
  buildInputs = optional pkgs.stdenv.isDarwin pkgs.openssl;
  # ...
}
```

**Comment by FRidh** [comment]
https://github.com/psycopg/psycopg2/blob/4e92322d7498fec1d48411836aaa981e968cd7f5/setup.py#L363
seems to explicitly add `ssl` as a required library for the extension type when the system is Darwin.

**Comment by copumpkin** [comment]
Oh, I guess that explains the difference :smile: I guess we can just add the conditional I wrote above

**Comment by copumpkin** [comment]
Fixed in df27b34918c03ffca62090c391ea159002e6ab9d.
